packages/blade/src/components/Box/BaseBox/types/propsTypes.ts

```diff
@@ -117,7 +131,11 @@ type BaseBoxVisualProps = MakeObjectResponsive<
 
 type BoxVisualProps = MakeObjectResponsive<{
   backgroundColor: BackgroundColorString<'surface'>;
-}>;
+}> & {
+  // Intentionally keeping this outside of MakeObjectResponsive since we only want as to be string and not responsive object
+  // styled-components do not support passing `as` prop as an object
+  as: BoxAsType;
```

Comment:
We can't do `as={{ base: 'section', m: 'footer' }}`. Styled Components is failing. The `as` property is not handled anywhere by us. Styled Component takes care of it.

Reply:
ye to anyways nai hona chaiye lol 🤣 

seems like an antipattern